File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int ans=0;
        long sumt=0;
        HashMap<Long, Integer> map = new HashMap<>();
        map.put((long)0, 1);
        for(int n: nums){
            sumt+=n;
            ans+=map.getOrDefault(sumt-k, 0);
            map.put(sumt, map.getOrDefault(sumt, 0)+1);
        }
        return ans;
    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        ans=sumt=0
        hm=defaultdict(int)
        hm[0]+=1
        for i in nums:
            sumt+=i
            ans+=hm[sumt-k]
            hm[sumt]+=1
        return ans